generate-vendor-tarball.sh

```
@@ -26,6 +33,16 @@ check_required_tools() {
 
 check_required_tools
 
+# Dependency patches (0001-0019) applied to the vendored source tree.
+# These are Cargo.toml patches that affect dependency resolution and therefore
+# determine which crates need to be vendored. Feature flag selection is handled
+# via --no-default-features --features passed to cargo vendor-filterer directly.
+# Discovered automatically by glob so adding/removing a patch file is enough.
+PATCHES=()
+while IFS= read -r patch; do
+    PATCHES+=("$patch")
+done < <(ls 000[0-9]-*.patch 001[0-9]-*.patch 2>/dev/null | sort)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)